pkg/configuration/converters.go

```diff
@@ -0,0 +1,169 @@
+package configuration
+
+import (
+	"digger/pkg/core/models"
+	"digger/pkg/utils"
+	"fmt"
+	"github.com/jinzhu/copier"
+	"path/filepath"
+)
+
+func copyProjects(projects []*ProjectYaml) []Project {
+	result := make([]Project, len(projects))
+	for i, p := range projects {
+		item := Project{p.Name,
+			p.Dir,
+			p.Workspace,
+			p.Terragrunt,
+			p.Workflow,
+			p.IncludePatterns,
+			p.ExcludePatterns,
+		}
+		result[i] = item
+	}
+	return result
+}
+
+func copyTerraformEnvConfig(terraformEnvConfig *TerraformEnvConfigYaml) *TerraformEnvConfig {
+	if terraformEnvConfig == nil {
+		return &TerraformEnvConfig{}
+	}
+	result := TerraformEnvConfig{}
+	result.State = make([]EnvVar, len(terraformEnvConfig.State))
+	result.Commands = make([]EnvVar, len(terraformEnvConfig.Commands))
+
+	for i, s := range terraformEnvConfig.State {
+		item := EnvVar{
+			s.Name,
+			s.ValueFrom,
+			s.Value,
+		}
+		result.State[i] = item
+	}
+	for i, s := range terraformEnvConfig.Commands {
+		item := EnvVar{
+			s.Name,
+			s.ValueFrom,
+			s.Value,
+		}
+		result.Commands[i] = item
+	}
+
+	return &result
+}
+
+func copyStage(stage *StageYaml) *models.Stage {
+	result := models.Stage{}
+	result.Steps = make([]models.Step, len(stage.Steps))
+
+	for i, s := range stage.Steps {
+		item := models.Step{
+			Action:    s.Action,
+			Value:     s.Value,
+			ExtraArgs: s.ExtraArgs,
+			Shell:     s.Shell,
+		}
+		result.Steps[i] = item
+	}
+	return &result
+}
+
+func copyWorkflowConfiguration(config *WorkflowConfigurationYaml) *WorkflowConfiguration {
+	result := WorkflowConfiguration{}
+	result.OnPullRequestClosed = make([]string, len(config.OnPullRequestClosed))
+	result.OnPullRequestPushed = make([]string, len(config.OnPullRequestPushed))
+	result.OnCommitToDefault = make([]string, len(config.OnCommitToDefault))
+
+	result.OnPullRequestClosed = config.OnPullRequestClosed
+	result.OnPullRequestPushed = config.OnPullRequestPushed
+	result.OnCommitToDefault = config.OnCommitToDefault
+	return &result
+}
+
+func copyWorkflows(workflows map[string]*WorkflowYaml) map[string]Workflow {
+	result := make(map[string]Workflow, len(workflows))
+	for i, w := range workflows {
+		envVars := copyTerraformEnvConfig(w.EnvVars)
+		plan := copyStage(w.Plan)
+		apply := copyStage(w.Apply)
+		configuration := copyWorkflowConfiguration(w.Configuration)
+		item := Workflow{
+			envVars,
+			plan,
+			apply,
+			configuration,
+		}
+		copier.Copy(&w, &item)
```

Comment:
do you need to do both Copy and manuall assigning variables?

Reply:
oops, I think it's merge mistake, that copier lib didn't even work for me